components/elm/src/biogeophys/CanopyFluxesMod.F90

```diff
@@ -842,18 +853,25 @@ subroutine CanopyFluxes(bounds,  num_nolakeurbanp, filter_nolakeurbanp, &
 
             !! Sakaguchi changes for stability formulation ends here
 
-            rah(p,2) = 1._r8/(csoilcn*uaf(p))
-            raw(p,2) = rah(p,2)
+            rah(p,below_canopy) = 1._r8/(csoilcn*uaf(p))
+            raw(p,below_canopy) = rah(p,below_canopy)
             if (use_lch4) then
-               grnd_ch4_cond(p) = 1._r8/(raw(p,1)+raw(p,2))
+               grnd_ch4_cond(p) = 1._r8/(raw(p,above_canopy)+raw(p,below_canopy))
             end if
 
             ! Stomatal resistances for sunlit and shaded fractions of canopy.
             ! Done each iteration to account for differences in eah, tv.
 
-            svpts(p) = el(p)                         ! pa
-            eah(p) = forc_pbot(t) * qaf(p) / 0.622_r8   ! pa
+            svpts(p) = el(p)                         ! Pa
+            eah(p) = forc_pbot(t) * qaf(p) / 0.622_r8   ! Pa
             rhaf(p) = eah(p)/svpts(p)
+
+            ! variables for history fields
+            rah_above(p)  = rah(p,above_canopy)
+            raw_above(p)  = raw(p,above_canopy)
+            rah_below(p)  = rah(p,below_canopy)
+            raw_below(p)  = raw(p,below_canopy)
+            vpd(p)        = max((svpts(p) - eah(p)), 50._r8) * 0.001_r8 ! kPa
```

Comment:
How about defining `50._r8` as a parameter?

Reply:
and maybe consider adding a comment that `0.001_r8` is for converting Pa to kPa.

Reply:
Good points, I will push a PR shortly with these changes.